language/tools/move-cli/src/utils/movey_credential.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context, Result};
+use move_command_line_common::movey_constants::MOVEY_URL;
+use std::fs;
+use toml_edit::easy::Value;
+
+pub const MOVEY_CREDENTIAL_PATH: &str = "/movey_credential.toml";
```

Comment:
This PR partially overlaps with https://github.com/move-language/move/pull/226

For example the `MOVEY_CREDENTIAL_PATH` is already defined there in a different file and manipulating of `movey_credential.toml` is also at least partially duplicated which makes it difficult to review these side-by-side.

My suggestion is that you either make this PR into a stacked diff on top of https://github.com/move-language/move/pull/226 to refactor the code shared between the two or we simply land https://github.com/move-language/move/pull/226 first and you redo this one on top of HEAD. Please let me know what you'd prefer to do (and on my side I will try to act as quickly as I can on the subsequent changes).

Reply:
Hi [awelc](https://github.com/awelc), good to see you.
Our goal is to get the service going so please merge #226 first and we will refactor #227 and #318 accordingly

Reply:
> Hi [awelc](https://github.com/awelc), good to see you. Our goal is to get the service going so please merge #226 first and we will refactor #227 and #318 accordingly

I reviewed #226 but it still requires some work - that is why I offered an option of creating a stacked diff for your consideration. In particular, I don't think it will run on Windows which is important as a large portion of Move developers seems to work on Windows as indicated by our internal surveys (and we plan to enable Windows CI builds soon for the core Move repo).

Reply:
In that case, we will look into #226 and make it Windows compatible, particularly with the location of the credential file.
In the meantime, please merge #227 as that's the more important PR.

Reply:
> In that case, we will look into #226 and make it Windows compatible, particularly with the location of the credential file. In the meantime, please merge #227 as that's the more important PR.

Both PRs suffer from similar problems (e.g., in terms of concatenating paths that in my opinion will not work on Windows), so neither can be merged right away. I did a review of #277, and we can reverse the order and try to land #277 first (and merge #266 on top of it later, trying to remove existing code duplication), but I will not be able to work on these any more tonight.

Also, I tried upload and tried to upload a package (https://github.com/MystenLabs/sui/tree/main/sui_programmability/examples/move_tutorial) and apparently it was uploaded successfully but I can't find it in Movey...
```
~/east-login-move/target/debug/move movey-upload 
Your package has been successfully uploaded to Movey
```

Reply:
Thanks [awelc](https://github.com/awelc), we will work on these and have them ready within today. It's Friday for us so we won't be able to response to you tomorrow.
About your uploaded package we believed you uploaded to staging instead of production.
It's here https://movey-app-staging.herokuapp.com/packages/MyFirstPackage
since we read from the move.toml file the package name won't be the same as your folder name.

Reply:
> Thanks [awelc](https://github.com/awelc), we will work on these and have them ready within today. It's Friday for us so we won't be able to response to you tomorrow. 

Please make sure that it runs on Windows (I will try to re-test, but it would be good if you did it yourselves).

> About your uploaded package we believed you uploaded to staging instead of production. It's here https://movey-app-staging.herokuapp.com/packages/MyFirstPackage since we read from the move.toml file the package name won't be the same as your folder name.

Does it mean that the data hard-coded in the source points to staging and will have to be changed in the future. I was a little worried about all this data being hard-coded and now I am worried even more...

Reply:
The movey credentials are here https://github.com/move-language/move/blob/62f645c4381848f42c4b03fc3a6d510d8a0263ff/language/move-command-line-common/src/movey_constants.rs
Since you are building under debug mode it defaulted to the staging endpoint. Under prod it will point to Movey.net so I don't think you need to be worried over hardcoded values.

Reply:
> The movey credentials are here https://github.com/move-language/move/blob/62f645c4381848f42c4b03fc3a6d510d8a0263ff/language/move-command-line-common/src/movey_constants.rs Since you are building under debug mode it defaulted to the staging endpoint. Under prod it will point to Movey.net so I don't think you need to be worried over hardcoded values.

Cool, makes perfect sense! Thank you for the explanation!

Reply:
According to our [tests](https://github.com/move-language/move/pull/226#discussion_r939951295), everything is working and up to date. Both PRs are designed to work independently. We will rebase once either is merged.